avocado_vt/test.py

```diff
@@ -330,43 +332,18 @@ def _runTest(self):
                             "as root may produce unexpected results!!!")
             logging.warning("")
 
-        # Find the test
-        subtest_dirs = []
         test_filter = bootstrap.test_filter
-
-        other_subtests_dirs = params.get("other_tests_dirs", "")
-        for d in other_subtests_dirs.split():
-            # If d starts with a "/" an absolute path will be assumed
-            # else the relative path will be searched in the bin_dir
-            subtestdir = os.path.join(self.bindir, d, "tests")
-            if not os.path.isdir(subtestdir):
-                raise exceptions.TestError("Directory %s does not "
-                                           "exist" % subtestdir)
-            subtest_dirs += data_dir.SubdirList(subtestdir,
-                                                test_filter)
-
+        subtest_dirs = utils.find_subtest_dirs(params.get("other_tests_dirs", ""),
+                                               self.bindir,
+                                               test_filter)
         provider = params.get("provider", None)
 
         if provider is None:
-            # Verify if we have the correspondent source file for
-            # it
-            generic_subdirs = asset.get_test_provider_subdirs(
-                'generic')
-            for generic_subdir in generic_subdirs:
-                subtest_dirs += data_dir.SubdirList(generic_subdir,
-                                                    test_filter)
-            specific_subdirs = asset.get_test_provider_subdirs(
-                params.get("vm_type"))
-            for specific_subdir in specific_subdirs:
-                subtest_dirs += data_dir.SubdirList(
-                    specific_subdir, bootstrap.test_filter)
+            subtest_dirs += utils.find_generic_specific_subtest_dirs.params.get(
+                "vm_type", test_filter)
```

Comment:
To improve readability and flexibility, isn't it better to break this function into two? `find_generic_subtest_dirs` and `find_specific_subtest_dirs`

Reply:
Actually, you've spotted a bug/typo.  There should be an opening parenthesis *before* `params.get`.

Reply:
I mean, it should be:

```python
            subtest_dirs += utils.find_generic_specific_subtest_dirs(
                params.get("vm_type"), test_filter)
```

Force pushing the fixes.

Reply:
> To improve readability and flexibility, isn't it better to break this function into two? `find_generic_subtest_dirs` and `find_specific_subtest_dirs`

Yes, I plan to do some extra refactoring here, and even some drastic changes later on, but I thought it would make sense to start smaller.